scopes/ui-foundation/ui/ui.main.runtime.ts

```diff
@@ -462,15 +471,23 @@ export class UiMain {
   async buildUiHash(uiRoot: UIRoot, runtime = 'ui'): Promise<string> {
     const aspects = await uiRoot.resolveAspects(runtime);
     aspects.sort((a, b) => (a.aspectPath > b.aspectPath ? 1 : -1));
-    const hash = aspects.map((aspect) => {
+    const aspectPathStrings = aspects.map((aspect) => {
       return [aspect.aspectPath, aspect.runtimePath].join('');
     });
-    return sha1(hash.join(''));
+    return sha1(aspectPathStrings.join(''));
   }
 
   async buildIfChanged(name: string, uiRoot: UIRoot, force: boolean | undefined): Promise<string> {
     this.logger.debug(`buildIfChanged, name ${name}`);
     const hash = await this.buildUiHash(uiRoot);
+
+    const preBundleHash = getPreBundleHash();
+    if (hash === preBundleHash && !force) {
+      this._isUiPreBundleServed = true;
+      this.logger.debug(`buildIfChanged, name ${name}, returned from pre-bundle`);
+      return hash;
+    }
```

Comment:
If someone runs bit start with --rebuild (force) then we generate the public and serve it from there.
but then if he stopped the server and starts again (without --rebuild) you will serve it from the pre-bundle even if he has it locally on the public folder in the ws with the same hash.
I would expect in case that you have the same hash already built locally to use it and not the pre-bundle.
(I'm not entirely sure that this is the case with the current impl)